docs/content/guide/interpolation.ngdoc

```diff
@@ -0,0 +1,125 @@
+@ngdoc overview
+@name Interpolation
+@sortOrder 275
+@description
+
+Interpolation markup with embedded @link {guide/expressions expressions} is used by Angular to
+provide data-binding to text nodes and attribute values.
+
+An example of interpolation is shown below:
+
+```html
+<a ng-href="img/{{username}}.jpg">Hello {{username}}!</a>
+```
+
+### How text and attribute bindings work
+
+During the compilation process the {@link ng.$compile compiler} uses the {@link ng.$interpolate $interpolate}
+service to see if text nodes and element attributes contain interpolation markup with embedded expressions.
+If that is the case, the compiler adds an interpolationDirective to the node and
+registers {@link ng.$rootScope.Scope#$watch watches} on the computed interpolation function,
+which will update the corresponding text nodes or attribute values as part of the
+normal {@link ng.$rootScope.Scope#$digest digest} cycle.
+
+### Binding to boolean attributes
```

Comment:
This section is very useful for sure, but I don't think it is related to interpolation.
If there isn't a better place to put it though, it's much better here than nowhere :)